developer/build_bundle_cache.py

```diff
@@ -0,0 +1,214 @@
+# Copyright (c) 2013 Shotgun Software Inc.
```

Comment:
nitpic: The help output says `_build_bundle_cache.py` in the examples

```
(toolkit_virtualenv)mannes-mbp:developer manne$ python build_bundle_cache.py -h
Usage: build_bundle_cache.py [options] config_descriptor target_path

Builds a bundle cache for a given configuration.

Options:
  -h, --help            show this help message and exit
  -d, --debug           Enable debug logging

  Shotgun Authentication:
    In order to download content from the Toolkit app store, the script
    will need to authenticate against any shotgun site. By default, it
    will use the toolkit authentication APIs stored credentials, and if
    such are not found, it will prompt for site, username and password.

    -s SHOTGUN_HOST, --shotgun-host=SHOTGUN_HOST
                        Shotgun host to authenticate with.
    -n SHOTGUN_SCRIPT_NAME, --shotgun-script-name=SHOTGUN_SCRIPT_NAME
                        Script to use to authenticate with the given host.
    -k SHOTGUN_SCRIPT_KEY, --shotgun-script-key=SHOTGUN_SCRIPT_KEY
                        Script key to use to authenticate with the given host.


Details and Examples
--------------------

In it's simplest form, provide a descriptor to a configuration and the location
where the bundle cache should be created.

> python _build_bundle_cache.py
            "sgtk:descriptor:app_store?version=v0.3.6&name=tk-config-basic"
            /tmp

Note that it is important to use quotes around the descriptor as shells usually
give special meaning to the & character.

For automated build setups, you can provide a specific shotgun API script name and
and corresponding script key:

> python _build_bundle_cache.py
            --shotgun-host='https://mysite.shotgunstudio.com'
            --shotgun-script-name='plugin_build'
            --shotgun-script-key='<script-key-here>'
            "sgtk:descriptor:app_store?version=v0.3.6&name=tk-config-basic" /tmp

For information about the various descriptors that can be used, see
http://developer.shotgunsoftware.com/tk-core/descriptor


```